dataclients/kubernetes/kube.go

```diff
@@ -467,11 +467,19 @@ func (c *Client) ingressToRoutes(items []*ingressItem) ([]*eskip.Route, error) {
 				if prule.Backend.Traffic > 0 {
 					r, err := c.convertPathRule(i.Metadata.Namespace, i.Metadata.Name, rule.Host, prule, servicesURLs)
 					if err != nil {
-						// tolerate single rule errors
+						// fail when there's a temporary network error so that we keep the last known rules
+						// active until the next sync loop.
+						if err, ok := err.(net.Error); ok {
+							log.Errorf("temporary error while getting service, deferring update: %v", err)
+							return nil, err
```

Comment:
> not good. This will delete all the existing routes on update if there is a network error after the initial load was successful.

Are you sure? If `LoadAll` or `LoadUpdate` fails it will check if `err != nil` and not do any route updates if that's the case, but just try again: https://github.com/zalando/skipper/blob/master/routing/datasource.go#L68-L80

I don't see why this would delete the existing routes.